src/download.rs

```diff
@@ -79,6 +80,66 @@ impl TryFrom<&str> for Download {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum Status {
+    Fail(String),
+    Success,
+}
+/// Represents a [`Download`] summary.
+#[derive(Debug, Clone)]
+pub struct Summary {
+    /// Downloaded items.
+    download: Download,
+    /// HTTP status code.
+    statuscode: StatusCode,
+    /// Download size in bytes.
+    size: u64,
+    /// Status.
+    status: Status,
+}
+
+impl Summary {
+    /// Create a new [`Download`] [`Summary`].
+    pub fn new(download: Download, statuscode: StatusCode, size: u64) -> Self {
+        Self {
+            download,
+            statuscode,
+            size,
+            status: Status::Fail("Unstarted".into()),
+        }
+    }
+
+    /// Attach a status to a [`Download`] [`Summary`].
+    pub fn with_status(self, status: Status) -> Self {
```

Comment:
You could also do something similar to how [`tonic::Status`](https://docs.rs/tonic/latest/tonic/struct.Status.html#implementations) propagates error codes here to simplify `summary.with_status(Status::Fail(e.to_string()))` calls:

```rs
pub fn fail(self, msg: impl std::fmt::Display) -> Self {
        Self {
            status: Status::Fail(format!("{}", msg)),
            ..self
        }
}
```